opentelemetry/proto/profiles/v1experimental/pprofextended.proto

```diff
@@ -82,6 +82,11 @@ message Profile {
   repeated Location location = 4;
   // Array of locations referenced by samples.
   repeated int64 location_indices = 15;
+  // Whether addresses within locations have been normalized to be
+  // execution-context agnostic. This is important, because a symbolizer needs
+  // this information to correctly process an address or not prior to
+  // symbolization.
+  bool normalized_addresses = 19;
```

Comment:
Should this be a profile, sample or location attribute? 

A profile may include samples of different types, and I'm unsure if normalized addresses  on this level places undue restrictions on the implementor.

Reply:
In theory, it could be a per-location thing, but the only examples we've seen in the wild are either everything in a profile or nothing since it's more of a "producer concern" than the location itself.

I agree describing the actual normalization scheme here makes sense. I'll add a link to the pprof base address calculation.

Reply:
Making it a `Location` attribute is probably excessive/too-fine-grained, what about making it a `Sample` attribute instead? 

This question has more to do with what we expect going forward, on the part of SDKs/implementors, since not a lot of folks are using this signal yet. If we make it a Profile-level attribute, then we'd be forcing everyone to split data with different semantics for frame addresses across Profile boundaries, thus missing out on e.g. unified string table.

@felixge @petethepig what do you think?

Reply:
Yeah, I see where you're coming from. Per sample would work as well, but of course adds additional bytes to each sample. I don't feel too strongly either way, I was just going to go with this since we've only ever seen it one way or the other for the whole profile, and I can't come up with why a counter-example would make sense.

Reply:
I'm not very familiar with address normalization, so apologies if this makes no sense, but wouldn't the `Mapping` be the right level for storing this?

Reply:
Address normalization is useful to make the address independent of the "real" address space of the process execution environment. This is helpful for saying things like "stack A on machine 1 is actually the same as stack B on machine 2 since their addresses match after normalization", which is good for compression and data bandwidth (when used in conjunction with stack IDs/hashing).

The information within a mapping is used for this, but it also essentially requires all ELF program headers of loadable section (I'd be very much for adding this optionally as well, but that should be a separate discussion I think).

Reply:
I agree with @felixge that storing this information within `message Mapping` should be preferred.

Reply:
There are two issues with that:

1. The addresses in question are not `Mapping` addresses, but `Location` addresses. So besides the semantic mismatch, if we make this a `Mapping` attribute, we risk confusion.
2. Granularity-wise, `Mapping` also feels the wrong place for this, in that we would never expect to see `Location` addresses with different normalization schemes dictated at the `Mapping` level. 

Semantically, `Location` is the best-fit but it may be too fine-grained for real-world use.

Reply:
Agreed with @christos68k. It describes a property of the Location's address.

I'd be ok with it being in the location, or if we think that's excessive in terms of space-use in the wire protocol for the sample or whole profile.

Reply:
Thanks for the proposal, I agree that normalizing addresses on the fly can make things simpler. It is not always easy to retrieve matching binaries. Having consistent addresses helps with investigations and offline symbolization.
Having a boolean at the level of the profile is slightly strange, as we will have inconsistent behaviors depending on binaries. 

Carrying this information at the level of the mapping is the most efficient way to store this information. We can either:

- Consider storing a boolean
The boolean would indicate if a normalization process was successfully done on the addresses stored in the locations (for the given mapping). Though I agree this is strange considering @christos68k ‘s comment.

- Store the result of the GetBase for this mapping to enable the computation of the normalized address. 
This still requires extra steps from the user to recompute the result of the normalization, though this ensures we have all the information to go from one address to another.

Putting the information at the level of the location seems slightly wasteful. I do not think it is possible to have cases where the normalization is different for locations within the same mapping. Although it does make things easier when looking at raw data (as you no longer need to use the adjustment from the mapping). In standard cases, we have ~1000s of locations, so this would add extra tens of kilobytes. If we start having millions of locations (aggregating over longer periods?unusual workloads?), this could become a reasonable optimization to consider.

Reply:
Thanks for the input @r1viollet! To summarize, the crystallized options that seem most appropriate (ignoring backend normalization) are:

1. Store a normalized address boolean in `Mapping`
2. Store a normalized address boolean in `Sample`

For 1., normalization state would apply to all `Locations` in `Mapping`.
For 2., normalization state would apply to all `Mappings` and `Locations` in `Sample`.

For both options we should document that if the boolean is `true`, indicating normalized addresses, `Mapping.memory_start` / `Mapping.memory_limit` could be missing or set to zero. It's not always convenient or even needed to collect these and this is one way to make them optional. That should address any possible confusion regarding these values.

If we don't expect to see (or need) `Mappings` with different normalization states in the same `Sample`, then 2. is slightly more efficient. I'm looking at this from the perspective of a continuous profiler that is in full control of extracting mappings and performing normalization, for which case it'd never need to change normalization state for `Mappings` in the same `Sample` (or even in the same `Profile`).

Reply:
Trying to think of use cases where normalization can go wrong in profiling, I think temporary extracted libraries could fall into this category:
You have native libraries that are created as temporary files, loaded, then deleted. You will find them with names like: `ibnetty_tcnative_linux_x86_12345.so` in java. in a tmp folder.
If you profile after the load, you would not find the elf file to perform the normalization (it would be flagged as deleted). Though offline, with the proper logic, you could symbolize using mappings and process level addresses. This could lead to a case where you would have inconsistent normalization states depending on mappings.

Reply:
It seems like the most alignment is on having it in Mapping, which I'm fine with, as at least it removes it to be repeated in every Location (although like I said above it actually more describes the Location's address but I don't feel too strongly about that).

> Store the result of the GetBase for this mapping to enable the computation of the normalized address.

Part of the whole point of not doing this is to ensure that the address is identical across hosts, other binaries that dynamically link a library, etc. as that opens the door to deduplication (both in the protocol and server side).